pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -225,6 +228,29 @@ func (m *Multicluster) AddMemberCluster(client kubelib.Client, clusterID string)
 		}
 	}
 
+	// setting up the serviceexport controller if and only if it is turned on in the meshconfig.
+	// TODO(nmittler): Need a better solution. Leader election doesn't take into account locality.
+	if features.EnableMCSServiceExport {
+		go leaderelection.
```

Comment:
Can you use `s.addTerminatingStartFunc(func(stop <-chan struct{}) error {` like the other usages? this ensures we can optimistically drop the leader election lock on shutdown, rather than waiting 30-60s for a new instance to pick it up

Reply:
This will require some refactoring, since the `Server` isn't accessible here. I can throw together a follow-on PR.

Reply:
Follow-on PR: https://github.com/istio/istio/pull/31521